File: provisioning/scim/search-list-users/code-samples/searchViaPost2.js
// https://developers.ringcentral.com/my-account.html#/applications
// Find your credentials at the above url, set them as environment variables, or enter them below

// POST BODY
const body = {
    count: 000,
    filter: '<ENTER VALUE>',
    schemas: [
        'urn:ietf:params:scim:api:messages:2.0:SearchRequest',
    ],
    startIndex: 000
};

const SDK = require('ringcentral');
const rcsdk = new SDK({server: process.env.serverURL, appKey: process.env.clientId, appSecret: process.env.clientSecret});
const platform = rcsdk.platform();
platform.login({ username: process.env.username, extension: process.env.extension, password: process.env.password }).then(() => {
    platform.post(`/scim/v2/Users/.search`, body).then((r) => {
        // PROCESS RESPONSE
    });
});

File: provisioning/scim/search-list-users/code-samples/searchViaPost2.cs
// https://developers.ringcentral.com/my-account.html#/applications
// Find your credentials at the above url, set them as environment variables, or enter them below

// POST BODY
SearchRequest searchRequest = new SearchRequest {
    count = 000,
    filter = "<ENTER VALUE>",
    schemas = new[] {
        'urn:ietf:params:scim:api:messages:2.0:SearchRequest',
    },
    startIndex = 000
};

RestClient rc = new RestClient(
    Environment.GetEnvironmentVariable("clientId"),
    Environment.GetEnvironmentVariable("clientSecret"),
    false
);
await rc.Authorize(
    Environment.GetEnvironmentVariable("username"),
    Environment.GetEnvironmentVariable("extension"),
    Environment.GetEnvironmentVariable("password")
);
var r = await rc.Scim().Users()..search().Post(searchRequest);
// PROCESS RESPONSE
